src/jsonpatcherproxy.js

```diff
@@ -81,32 +81,41 @@ const JSONPatcherProxy = (function() {
     Why do we need to check instance._isProxifyingTreeNow?
 
     We need to make sure we mark revocables as inherited ONLY when we're observing,
-    because throughout the first proxification, a sub-object is proxified and then assigned to 
+    because throughout the first proxification, a sub-object is proxified and then assigned to
     its parent object. This assignment of a pre-proxified object can fool us into thinking
-    that it's a proxified object moved around, while in fact it's the first assignment ever. 
+    that it's a proxified object moved around, while in fact it's the first assignment ever.
 
-    Checking _isProxifyingTreeNow ensures this is not happening in the first proxification, 
+    Checking _isProxifyingTreeNow ensures this is not happening in the first proxification,
     but in fact is is a proxified object moved around the tree
     */
     if (subtreeMetadata && !instance._isProxifyingTreeNow) {
       subtreeMetadata.inherited = true;
     }
 
+    let warnedAboutNonIntegrerArrayProp = false;
+    const isTreeAnArray = Array.isArray(tree);
+
     // if the new value is an object, make sure to watch it
     if (
       newValue &&
       typeof newValue == 'object' &&
       !instance._treeMetadataMap.has(newValue)
     ) {
-      instance._parenthoodMap.set(newValue, { parent: tree, key });
-      newValue = instance._proxifyTreeRecursively(tree, newValue, key);
+      if (isTreeAnArray && !Number.isInteger(+key.toString())) {
```

Comment:
cosmetic: Just for my personal readability, we could also create a `const isArrayProperty = isTreeAnArray && !Number.isInteger(+key.toString())` not to repeat it here and at l.139. Slightly DRYier vs. one `const` variable less. If you find it readable and better this way, let's keep it :)

Reply:
Done